script.js

```diff
@@ -48,3 +106,87 @@ function onSubmit() {
   const converted = amountIn * rate;
   amountOut.value = converted;
 }
+
+
+function searchFunction() {
+  // Declare variables
+  let input,
+    filter,
+    table,
+    tr,
+    td, td1, td2,
+    i,
+    txtValue,
+    txtValue1,
+    rateValue,
+    inputRateFrom,
+    inputRateTo;
+  input = document.getElementById("searchInp");
+  filter = input.value.toUpperCase();
+  table = document.getElementById("rateTable");
+  tr = table.getElementsByTagName("tr");
+
+  inputRateFrom = +(document.getElementById("searchRateFrom").value);
+  inputRateTo = +((document.getElementById("searchRateTo").value) === '' ?
+    100000000 : (document.getElementById("searchRateTo").value));
+  
+  
+  // Loop through all table rows, and hide those who don't match the search query
+  for (i = 0; i < tr.length; i++) {
+    td = tr[i].getElementsByTagName("td")[0];
+    td1 = tr[i].getElementsByTagName("td")[1];
+    td2 = tr[i].getElementsByTagName("td")[2];
+
+    if (td || td1 || td2) {
+      txtValue = td.textContent || td.innerText;
+      txtValue1 = td1.textContent || td1.innerText;
+      rateValue =+ td2.textContent;
+
+      if (inputRateFrom >= 0 || inputRateTo !== 0) {
+        if (rateValue >= inputRateFrom && rateValue <= inputRateTo) {
+          console.log("goes  in rate");
+          tr[i].style.display = "";
+        } else {
+          tr[i].style.display = "none";
+          console.log("out  of rate");
+        };
+      } else {
```

Comment:
haha, it was a bit tricky to enable search by name :D I needed to set `from` to -1 and `to` to 0 to make it work :D I love this easter egg :D  it happens because by default the from is set to `0` and to `100000000`, maybe it would help you instead of 1 search form you would have 2, then you do not need make this complicate conditions